apps/tangle-dapp/containers/DelegateTxContainer/DelegateTxContainer.tsx

```diff
@@ -49,6 +51,24 @@ const DelegateTxContainer: FC<DelegateTxContainerProps> = ({
     [getActiveValidators.name],
     ([, ...args]) => getActiveValidators(...args)
   );
+  const { data: waitingValidatorsData } = useSWR(
+    [getWaitingValidators.name],
+    ([, ...args]) => getWaitingValidators(...args)
+  );
+
+  const [maxNominationQuota, setMaxNominationQuota] = useState<number>(0);
+
+  useEffect(() => {
+    getMaxNominationQuota().then((maxNominationQuota) => {
+      setMaxNominationQuota(maxNominationQuota ? maxNominationQuota : 16);
```

Comment:
The variable `maxNominationQuota` can potentially be `undefined`, `0`, or `NaN`. I recommend implementing a more stringent check for this scenario.